File: sprint05/JMS/FW/PGS_42586.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int maxday = 0;
    
    for(int i=0;i<progresses.size();i++){
        int amount = 100 - progresses[i];
        int day = amount / speeds[i];
        
        if(amount % speeds[i] > 0)
            day++;
        
        if(day > maxday){
            maxday = day;
            answer.push_back(1);
        }
        else
            answer[answer.size()-1]++;
    }
    
    return answer;
}

File: sprint05/KKW/First week/PGS_42586.py
from collections import deque
def solution(progresses,speeds):
    answer=[]
    cnt=0
    day=1
    progresses=deque(progresses)
    speeds=deque(speeds)
    while len(progresses)>0:
        if progresses[0]+speeds[0]*day>=100:
            progresses.popleft()
            speeds.popleft()
            cnt+=1
        else:
            day+=1
            if cnt>0:
                answer.append(cnt)
                cnt=0
    answer.append(cnt)
    return answer